crates/prover/src/runner.rs

```diff
@@ -786,6 +800,55 @@ where
     pub fn get_state_root(&self) -> &Stf::StateRoot {
         &self.state_root
     }
+
+    fn break_sequencer_commitments_into_groups(
+        &self,
+        sequencer_commitments: Vec<SequencerCommitment>,
+    ) -> anyhow::Result<Vec<Vec<SequencerCommitment>>> {
+        let mut result = vec![];
+
+        let mut group = vec![];
+        let mut cumulative_state_diff = StateDiff::new();
+        for sequencer_commitment in sequencer_commitments {
+            group.push(sequencer_commitment.clone());
+            for l2_height in sequencer_commitment.l2_start_block_number
+                ..=sequencer_commitment.l2_end_block_number
+            {
+                let state_diff = self
+                    .ledger_db
+                    .get_l2_state_diff(BatchNumber(l2_height))?
+                    .ok_or(anyhow!(
+                        "Could not find state diff for L2 range {}-{}",
+                        sequencer_commitment.l2_start_block_number,
+                        sequencer_commitment.l2_end_block_number
+                    ))?;
+                cumulative_state_diff = merge_state_diffs(cumulative_state_diff, state_diff);
+            }
+
+            let serialized_state_diff = bincode::serialize(&cumulative_state_diff)?;
```

Comment:
we use borsh for this inside apply_sequencer_commitments_from_da I believe. so we should use that as well. pls double check

Reply:
bincode in this case is only used to serialize and then calculate the diff size. This is also used in the sequencer the same way. Do you want to switch to borsh?

Reply:
yes should have been borsh, probably we fdorgot to update those parts